solr/solr-ref-guide/src/distributed-requests.adoc

```diff
@@ -115,17 +115,19 @@ If specified, the thread pool will use a backing queue instead of a direct hando
 `fairnessPolicy`::
 Chooses the JVM specifics dealing with fair policy queuing, if enabled distributed searches will be handled in a First in First out fashion at a cost to throughput. If disabled throughput will be favored over latency. The default is `false`.
 
-`shardsWhitelist`::
+In addition, `HttpShardHandlerFactory` also depends on the following top-level property:
+
+`allowUrls`::
 If specified, this lists limits what nodes can be requested in the `shards` request parameter.
 +
-In SolrCloud mode this whitelist is automatically configured to include all live nodes in the cluster.
+In SolrCloud mode this allow-list is automatically configured to include all live nodes in the cluster.
 +
-In standalone mode the whitelist defaults to empty (sharding not allowed).
+In standalone mode the allow-list defaults to empty (sharding not allowed).
 +
-If you need to disable this feature for backwards compatibility, you can set the system property `solr.disable.shardsWhitelist=true`. The value of this parameter is a comma separated list of the nodes that will be whitelisted, i.e.,
+If you need to disable this feature for backwards compatibility, you can set the system property `solr.disable.allowUrls=true`. The value of this parameter is a comma separated list of the nodes that will be allowed, i.e.,
 `10.0.0.1:8983/solr,10.0.0.1:8984/solr`.
 +
-NOTE: In SolrCloud mode, if at least one node is included in the whitelist, then the `live_nodes` will no longer be used as source for the list. This means that if you need to do a cross-cluster request using the `shards` parameter in SolrCloud mode (in addition to regular within-cluster requests), you'll need to add all nodes (local cluster + remote nodes) to the whitelist.
+NOTE: In SolrCloud mode, if at least one node is included in the allow-list, then the `live_nodes` will no longer be used as source for the list. This means that if you need to do a cross-cluster request using the `shards` parameter in SolrCloud mode (in addition to regular within-cluster requests), you'll need to add all nodes (local cluster + remote nodes) to the allow-list.
```

Comment:
IMO this is a shame but I understand we don't want to change this approach in this PR.

Reply:
Yes, I don't understand neither. Is this a security matter? Indeed I prefer to handle that separately but that would be nice to open the discussion. Technically it is really easy to have the union of the allow-list and the live nodes.

Reply:
I don't know why it is what it is.  But once we want to start using this checker to check connecting to other hosts outside of the SolrCloud cluster, we'll then want to change this at that time.  Ideally this changes at 9.0 (major release boundary) so I suggest raising the question on the dev list now.

Reply:
I know, the local cluster’s live_nodes set should probably always be added to the manually configured allow-list. Should not be too hard?

Reply:
Ok, I have modified AllowListUrlChecker to always add the cluster live nodes to the configured allow-list. It actually reduces the code.